### DISCUSSION TITLE
Encrypted DB password in keycloak.conf file

### DISCUSSION THREAD

**Discussion by TorsonD** [keycloak/keycloak#12552] [Q&A]
Is it possible to store db password in keycloak.conf file in encrypted form?
From what i've read in ver 18 manuals, there is no such feature. Is it true?
Or is there some way to encrypt/mask it?
It is a bit confusing that security-related product would allow storing plaintext passwords in configuration files

**Comment by darius-m** [comment]
Is there any good way to protect / encrypt the password from an attacker with filesystem access equivalent to the Keycloak service?

As far as I can tell, encryption would require access to the decryption key when the service starts. Unless the system loses access to the key immediately after the Keycloak service starts, the attacker could still decrypt the password by accessing the key. This would probably mean that whenever the service goes down, the system would need to be reset in some manner.
